pkg/apis/pingcap/v1alpha1/types.go

```diff
@@ -104,62 +104,91 @@ type TidbClusterList struct {
 type TidbClusterSpec struct {
 
 	// PD cluster spec
-	PD PDSpec `json:"pd,omitempty"`
+	PD PDSpec `json:"pd"`
```

Comment:
Do not omit empty for mandatory fields, so that the it is more clear that this struct is wrongly empty if the validation is bypassed or not present.